central/image/service/http_handler.go

```diff
@@ -0,0 +1,153 @@
+		httputil.WriteGRPCStyleError(w, codes.NotFound, errors.New("Scanner v4 is not enabled. Enabled scanner v4 to get SBOMs"))
+		return
+	}
+	if !features.SBOMGeneration.Enabled() {
+		httputil.WriteGRPCStyleError(w, codes.NotFound, errors.New("SBOM feature is not enabled"))
+		return
+	}
+	var params sbomRequestBody
+	reqSizeLimit := NumberBytes
+	reqSizeBytes := os.Getenv("ROX_SBOM_API_REQ_SIZE")
+	if reqSizeBytes != "" {
+		reqSizeLimitInt, err := strconv.Atoi(reqSizeBytes)
+		if err == nil {
+			reqSizeLimit = reqSizeLimitInt
+		}
+	}
+	// timeout api after 10 minutes
+	ctx, cancel := context.WithTimeout(r.Context(), time.Duration(timeout))
+	defer cancel()
+	//for testing only
+	log.Infof("request size is %d", reqSizeLimit)
+	lr := io.LimitReader(r.Body, int64(reqSizeLimit))
+	err := json.NewDecoder(lr).Decode(&params)
+	if err != nil {
+		httputil.WriteGRPCStyleError(w, codes.InvalidArgument, errors.Wrap(err, "decoding json request body"))
+		return
+	}
+	bytes, err := h.getSbom(ctx, params)
+	if err != nil {
+		httputil.WriteGRPCStyleError(w, codes.Internal, errors.Wrap(err, "error generating SBOM"))
+		return
+	}
+
+	// Tell the browser this is a download.
+	w.Header().Add("Content-Disposition", "attachment; sbom.json")
+	w.Header().Add("Content-Type", "application/json")
+	w.Header().Add("Content-Length", fmt.Sprint(len(bytes)))
+	_, _ = w.Write(bytes)
+}
+
+func (h sbomHttpHandler) enrichImage(params sbomRequestBody, ctx context.Context) (*storage.Image, error) {
+	enrichmentCtx := enricher.EnrichmentContext{
+		FetchOpt:  enricher.UseCachesIfPossible,
+		Delegable: true,
+	}
+
+	if params.Force {
+		enrichmentCtx.FetchOpt = enricher.UseImageNamesRefetchCachedValues
+	}
+
+	if params.Cluster != "" {
+		// The request indicates enrichment should be delegated to a specific cluster.
+		clusterID, err := clusterUtil.GetClusterIDFromNameOrID(ctx, h.clusterSACHelper, params.Cluster, delegateScanPermissions)
+		if err != nil {
+			return nil, err
+		}
+		enrichmentCtx.ClusterID = clusterID
+	}
+
+	img, err := enricher.EnrichImageByName(ctx, h.enricher, enrichmentCtx, params.ImageName)
+	if err != nil {
+		return nil, err
+	}
+	return img, nil
+}
+
+func (h sbomHttpHandler) getSbom(ctx context.Context, params sbomRequestBody) ([]byte, error) {
+	// enrich image checks image metadata cache if fetchopt = UseCachesIfPossible otherwise fetches metdata from registry
+	// enrich image calls get scans on image which creates index report for image if it does not exsist
+	_, err := h.enrichImage(params, ctx)
+	if err != nil {
+		return nil, err
+	}
+	// how to verify enrichimage failed because index report not found
+	// get sbom from matcher
+	// for testing only
+	sbom := map[string]interface{}{
+		"SPDXID":      "SPDXRef-DOCUMENT",
+		"spdxVersion": "SPDX-2.3",
+		"creationInfo": map[string]interface{}{
+			"created": "2023-08-30T04:40:16Z",
+			"creators": []string{
+				"Organization: Uchiha Cortez",
+				"Tool: FOSSA v0.12.0",
```

Comment:
SUPER NIT: Just so this persons name is not in the master branch's git history, lets change it something else.

```suggestion
				"Organization: Acme Org",
				"Tool: N/A - PoC",
```

Reply:
done